### ISSUE TITLE
A question regarding the lr scheduler

### ISSUE THREAD

**Issue by viyjy** [mosaicml/diffusion#27]
Hi, is that correct to set milestones to 200ep? Thanks.
```
milestones:
    - 200ep
```
https://github.com/mosaicml/diffusion/blob/2eb6228ef205832aaf7d3111b64a38cdfa4c06f6/yamls/hydra-yamls/SD-2-base-256.yaml#LL62C14-L62C14

**Comment by Landanjs** [comment]
Hi, this is a hold over from a previous experiment. For our training LR schedule, we used linear warmup then a constant learning rate. As long as your number of epochs is < 200, the config you posted will work.

Below is an alternative more accurate config:

```
scheduler:
  _target_: composer.optim.LinearWithWarmupScheduler
  t_warmup: 10000ba
  alpha_i: 1.0
  alpha_f: 1.0
```

**Comment by viyjy** [comment]
Thanks. 
Can you provide the accurate SD-2-base-256.yaml and SD-2-base-512.yaml that were used in your experiment? So that we can reproduce the SD-2-base you have trained.

**Comment by Landanjs** [comment]
The yaml you referenced in the repo are the ones we used for training except for the missing dataset paths. 

The alternative scheduler I posted above was an adjustment we wanted to make, but did not have a chance to add it.

**Comment by viyjy** [comment]
So I should use the alternative scheduler to reproduce your model training, am I correct?

**Comment by Landanjs** [comment]
Apologies for the confusion. The config we used to train was the one you specified above: 
```
milestones:
    - 200ep
```

The alternative scheduler should be equivalent and we plan to use it for future runs, but **we did not use the alternative scheduler for our blog post training run**.

**Comment by viyjy** [comment]
it's actually the same, right? since the training won't reach to 200ep, so the lr will be the constant value=1e-4 after warmup

**Comment by viyjy** [comment]
BTW, does this yaml file contain all those [five techniques](https://www.mosaicml.com/blog/diffusion) to speed up the training? Or should I add them by myself? Thanks~

**Comment by Landanjs** [comment]
Yes, they are the same.

All five techniques should be enabled in the yaml. A caveat is that precomputing the VAE and CLIP latents must be done before training (if you want to use precomputed latents).
